fileshare/event_manager_test.go

```diff
@@ -1363,3 +1374,132 @@ func TestAutoaccept(t *testing.T) {
 		})
 	}
 }
+
+func TestEventsFlow(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	const (
+		numEvents  uint64 = 1024
+		chunkSize  uint64 = 8
+		chunkCount uint64 = numEvents / chunkSize
+		transferID string = exampleUUID
+		peer              = "12.12.12.12"
+		path              = tmpDir
+		file              = "testfile"
+		fileID            = "testfileid"
+		fileSize          = numEvents * 100
+	)
+
+	events := []Event{
+		{
+			Kind: EventKindRequestQueued{
+				Peer:       peer,
+				TransferId: transferID,
+				Files: []QueuedFile{
+					{
+						Id:   fileID,
+						Path: file,
+						Size: fileSize,
+					},
+				},
+			},
+		},
+		{
+			Kind: EventKindFileStarted{
+				TransferId: transferID,
+				FileId:     fileID,
+			},
+		},
+	}
+
+	// subtract by the existing RequestQueued and FileStarted events
+	// to make sure we have the correct number of events
+	// otherwise it's not chunked properly, as the size is not disivable by chunkSize(1026 % 8 = 2)
```

Comment:
[spelling] `disivable`
Add an assert into the test if the `events` length is not exactly divisible with `8`.
So if the tests is changed in the future to know sooner why it would fail.